api/v1beta3/auth_config_types.go

```diff
@@ -614,6 +614,9 @@ type KubernetesSubjectAccessReviewAuthorizationSpec struct {
 	// Groups the user must be a member of or, if `user` is omitted, the groups to check for authorization in the Kubernetes RBAC.
 	Groups []string `json:"groups,omitempty"`
 
+	// AuthorizationGroups is a value or selector to use as groups to check for authorization in the Kubernetes RBAC.
+	AuthorizationGroups *ValueOrSelector `json:"authorizationGroups,omitempty"`
```

Comment:
Perhaps we should deprecate `groups`, as `authorizationGroups` suffices, covering as well for the case of static values.

Reply:
I agree, we can deprecate `groups`. What's the process for doing so in Authorino? Is it just documenting it as deprecated?

Reply:
Right now, just adding the deprecation notice to the description of the field is good enough. I.e.

```go
	// Groups the user must be a member of or, if `user` is omitted, the groups to check for authorization in the Kubernetes RBAC.
	// Deprecated: Use authorizationGroups instead.
	Groups []string `json:"groups,omitempty"`
```

Reply:
I added the suggested doc string.